src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-views/use-psbt-decoded-request.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import * as btc from '@scure/btc-signer';
+
+import { BitcoinNetworkModes } from '@shared/constants';
+import { getAddressFromOutScript } from '@shared/crypto/bitcoin/bitcoin.utils';
+import { logger } from '@shared/logger';
+import { isUndefined } from '@shared/utils';
+
+import {
+  OrdApiInscriptionTxOutput,
+  useOrdinalsAwareUtxoQueries,
+} from '@app/query/bitcoin/ordinals/ordinals-aware-utxo.query';
+import { useCurrentBtcNativeSegwitAccountAddressIndexZero } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
+
+import { buildPsbtInputsForUi } from './build-psbt-input-for-ui';
+import { matchInputsOutputs } from './match-inputs-and-outputs';
+
+interface NonWitnessUtxo {
+  version: number;
+  segwitFlag: boolean;
+  inputs: btc.TransactionInputRequired[];
+  outputs: btc.TransactionOutputRequired[];
+  witnesses: string[][];
+  lockTime: number;
+}
+
+interface WitnessUtxo {
+  script: Uint8Array;
+  amount: bigint;
+}
```

Comment:
Can we pull from the lib?

Reply:
Nothing was exported that I saw that is why these were created. I can check again, but not a lot is exported.

Reply:
Prob going to have to build out a lot more with types here to handles all possible that could be passed? Signer lib has a lot of code handling it but doesn't export to use.